packages/core/src/shared/filetypes.ts

```diff
@@ -349,5 +357,6 @@ export const codefileExtensions = new Set([
 /** Returns true if `filename` is a code file. */
 export function isCodeFile(filename: string): boolean {
     const ext = path.extname(filename).toLowerCase()
-    return codefileExtensions.has(ext)
+    const result = codefileExtensions.has(ext) || codefileExtensions.has(filename)
```

Comment:
Can you elaborate on why we need `codefileExtensions.has(filename)`?

Reply:
`.gitignore` for example is ignored 

`path.extname(".gitignore")` is unable to return `gitignore`, it actually just returns nothing